src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -346,22 +345,38 @@ private ImmutableArray<ISymbol> GetSymbolsForExpressionOrStatementContext()
             var filterOutOfScopeLocals = _filterOutOfScopeLocals;
             if (filterOutOfScopeLocals)
             {
-                var contextNode = _context.LeftToken.GetRequiredParent();
                 filterOutOfScopeLocals =
                     !contextNode.IsFoundUnder<LocalDeclarationStatementSyntax>(d => d.Declaration.Type) &&
                     !contextNode.IsFoundUnder<DeclarationExpressionSyntax>(d => d.Type);
             }
 
-            var symbols = !_context.IsNameOfContext && _context.LeftToken.GetRequiredParent().IsInStaticContext()
-                ? _context.SemanticModel.LookupStaticMembers(_context.LeftToken.SpanStart)
-                : _context.SemanticModel.LookupSymbols(_context.LeftToken.SpanStart);
+            ImmutableArray<ISymbol> symbols;
+            if (_context.IsNameOfContext)
+            {
+                symbols = semanticModel.LookupSymbols(_context.LeftToken.SpanStart);
+
+                // We may be inside of a nameof() on a method.  In that case, we want to include the parameters in
+                // the nameof if LookupSymbols didn't already return them.
+                var enclosingMethodOrLambdaNode = contextNode.AncestorsAndSelf().FirstOrDefault(n => n is AnonymousFunctionExpressionSyntax or BaseMethodDeclarationSyntax);
+                var enclosingMethodOrLambda = enclosingMethodOrLambdaNode is null
+                    ? null
+                    : semanticModel.GetSymbolInfo(enclosingMethodOrLambdaNode).GetAnySymbol() ?? semanticModel.GetDeclaredSymbol(enclosingMethodOrLambdaNode);
+                if (enclosingMethodOrLambda is IMethodSymbol method)
+                    symbols = [.. symbols.Concat(method.Parameters)];
```

Comment:
>[..  [](http://example.com/codeflow?start=30&length=4)

is this the same as 

symbols = symbols.Concat(method.Parameters);

Reply:
likely yes.  will fixup.

Reply:
That's confusing. symbols is an IA, method.Parameters is an IA, isn't this invoking the IA overload for Concat?

Reply:
no, it's not.  i get an error when i try :)

Reply:
oh, I see. One is IA<ISymbol>, the other is IA<IParamaterSymbol>. It needs to be:

symbols.Concat(method.Parameters.Cast<ISymbol>());

Reply:
oh, just a sec, I thought that cast call just cast the array, not enumerate

Reply:
Duh, it's CastArray, not Cast